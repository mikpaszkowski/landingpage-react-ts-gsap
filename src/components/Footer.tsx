import * as React from "react";
import { Logo } from "./Logo";
import styled from "styled-components";
import { colors, fontTypes } from "../styles/styleConstants";

const FooterWrapper = styled.div`
  position: relative;
  width: 100%;
  padding: 8.9rem 16.5rem 4.7rem 16.5rem;
  background-color: ${colors.footerBackground};
`;

const BreakLine = styled.div`
  height: 0.1rem;
  background-color: white;
  width: 100%;
  opacity: 0.2;
`;

const TermsAndConditions = styled.p`
  position: absolute;
  bottom: 1.6rem;
  right: 16.6rem;
  font-family: ${fontTypes.secondaryFont};
  color: ${colors.primaryLight};
  opacity: 0.5;
`;

export const Footer = () => {
  return (
    <FooterWrapper>
      <Logo inverse />
      <BreakLine />
      <TermsAndConditions>Therms & Conditions</TermsAndConditions>
    </FooterWrapper>
  );
};
