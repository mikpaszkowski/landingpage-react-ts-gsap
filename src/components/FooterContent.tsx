import * as React from "react";

export const FooterContent = () => {};
