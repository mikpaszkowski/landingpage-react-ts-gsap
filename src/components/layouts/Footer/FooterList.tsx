import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../../styles/constants";
import { lists } from "../../staticContents/footer/footerLists";
import { device } from "../../../styles/responsive";

const List = styled.ul`
  & > li {
    color: ${({ theme }) => theme.palette.primary.fontLight};
    font-family: ${fontTypes.secondaryFont};
    font-style: normal;
    font-weight: normal;
    font-size: 1.8rem;
    line-height: 2.7rem;
    list-style-type: none;

    &:first-child {
      color: ${({ theme }) => theme.palette.secondary.fontLight};
      margin-bottom: 1.9rem;
    }
  }

  @media ${device.mobileMedium} {
    margin-bottom: 3.1rem;
  }
`;

interface FooterListProps {
  listName: string;
}

export const FooterList = ({ listName }: FooterListProps) => {
  return (
    <List>
      <li>{listName}</li>
      {lists[`${listName.toLocaleLowerCase()}`].map((item, index) => (
        <li key={index}>{item}</li>
      ))}
    </List>
  );
};
