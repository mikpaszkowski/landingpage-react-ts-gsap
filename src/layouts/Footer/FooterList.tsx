import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/constants";
import { lists } from "../../data/staticContents/footer/footerLists";
import { device } from "../../styles/responsive";

const List = styled.ul`
  margin-bottom: 3.1rem;
  & > li {
    font-family: ${fontTypes.secondaryFont};
    font-style: normal;
    font-weight: normal;
    font-size: 1.8rem;
    line-height: 2.7rem;
    list-style-type: none;

    & a {
      color: ${({ theme }) => theme.palette.primary.fontLight};
      text-decoration: none;
      font-size: 1.8rem;
      line-height: 2.7rem;
    }

    &:first-child {
      text-transform: uppercase;
      color: ${({ theme }) => theme.palette.secondary.fontLight};
      margin-bottom: 1.9rem;
    }
  }

  @media ${device.mobileMedium} {
    margin-bottom: 3.1rem;
  }
`;

interface IFooterListProps {
  listName: string;
}

export const FooterList = ({ listName }: IFooterListProps) => {
  return (
    <List>
      <li>{listName}</li>
      {lists[`${listName.toLocaleLowerCase()}`].map(({ url, label }, index) => (
        <li key={index}>
          <a href={url}>{label}</a>
        </li>
      ))}
    </List>
  );
};
