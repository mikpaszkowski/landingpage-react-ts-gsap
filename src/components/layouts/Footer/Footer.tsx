import * as React from "react";
import { Logo } from "../../ReusableComponents/Logo/Logo";
import styled from "styled-components";
import { fontTypes } from "../../../styles/styleConstants";
import { FooterContent } from "./FooterContent";
import { device } from "../../../styles/responsive";

const FooterWrapper = styled.div`
  position: relative;
  width: 100%;
  padding: 8.9rem 16.5rem 4.7rem 16.5rem;
  background-color: ${({ theme }) => theme.palette.primary.footerBackground};

  @media ${device.mobileMedium} {
    padding: 6.4rem 2.4rem 5.5rem 2.4rem;
  }
`;

const BreakLine = styled.div`
  height: 0.1rem;
  background-color: white;
  width: 100%;
  opacity: 0.2;

  @media ${device.mobileMedium} {
    display: none;
  }
`;

const TermsAndConditions = styled.p`
  position: absolute;
  bottom: 1.6rem;
  right: 16.6rem;
  font-family: ${fontTypes.secondaryFont};
  color: ${({ theme }) => theme.palette.primary.fontLight};
  opacity: 0.5;

  @media ${device.mobileMedium} {
    display: none;
  }
`;

export const Footer = () => {
  return (
    <FooterWrapper>
      <Logo inverse />
      <FooterContent />
      <BreakLine />
      <TermsAndConditions>Therms & Conditions</TermsAndConditions>
    </FooterWrapper>
  );
};
