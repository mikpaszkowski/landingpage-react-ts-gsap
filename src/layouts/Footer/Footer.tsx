import * as React from "react";
import { Logo } from "../../components/Logo/Logo";
import styled from "styled-components";
import { fontTypes } from "../../styles/constants";
import { FooterContent } from "./FooterContent";
import { device } from "../../styles/responsive";

const FooterWrapper = styled.div`
  position: relative;
  width: 100%;
  padding: 8.9rem 16.5rem 4.7rem 16.5rem;
  background-color: ${({ theme }) => theme.palette.primary.footerBackground};

  @media ${device.tablet} {
    padding: 6.4rem 2.4rem 5.5rem 2.4rem;
  }
`;

const BreakLine = styled.div`
  display: block;
  height: 0.1rem;
  background-color: white;
  width: 100%;
  opacity: 0.2;

  @media ${device.mobileLarge} {
    display: none;
  }
`;

const TermsAndConditions = styled.p`
  display: block;
  position: absolute;
  bottom: 1.6rem;
  right: 16.6rem;
  font-family: ${fontTypes.secondaryFont};
  color: ${({ theme }) => theme.palette.primary.fontLight};
  opacity: 0.5;

  @media ${device.mobileLarge} {
    display: none;
  }

  @media ${device.tablet} {
    right: 2.5rem;
  }
`;

export const Footer = () => {
  return (
    <FooterWrapper>
      <Logo inverse />
      <FooterContent />
      <BreakLine />
      <TermsAndConditions>Therms & Conditions</TermsAndConditions>
    </FooterWrapper>
  );
};
