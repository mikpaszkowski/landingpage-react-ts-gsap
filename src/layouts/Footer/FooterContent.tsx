import * as React from "react";
import styled from "styled-components";
import { FooterList } from "./FooterList";
import { EmailInputContainer } from "../../components/EmailInputContainer";
import { device } from "../../styles/responsive";

const FooterContentWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
  margin: 5.1rem 0 8rem 0;

  @media ${device.tabletLarge} {
    flex-direction: column-reverse;
    justify-content: flex-start;
    margin: 5.6rem 0 0 0;
  }
`;

const FooterListWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;

  ul:nth-child(2) {
    margin-right: 14rem;
  }

  ul:first-child {
    margin-right: 5.8rem;
  }

  @media ${device.tabletLarge} {
    flex-direction: column;
    justify-content: flex-start;
  }
`;

const footerLists = [
  {
    id: 1,
    listName: "products",
  },
  {
    id: 2,
    listName: "company",
  },
  {
    id: 3,
    listName: "help",
  },
];

export const FooterContent = () => {
  return (
    <FooterContentWrapper>
      <FooterListWrapper>
        {footerLists.map(({ id, listName }) => (
          <FooterList key={id} listName={listName} />
        ))}
      </FooterListWrapper>
      <EmailInputContainer />
    </FooterContentWrapper>
  );
};
