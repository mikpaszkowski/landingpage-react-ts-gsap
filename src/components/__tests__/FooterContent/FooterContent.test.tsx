import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithLightTheme, mountWithDarkTheme } from "../helpers/index";
import { FooterContent } from "../../layouts/Footer/FooterContent";

afterEach(() => {
  cleanup();
});

test("check render correctly <FooterContent /> component", () => {
  const tree = renderWithLightTheme(<FooterContent />).toJSON();
  expect(tree).toMatchSnapshot();
});
