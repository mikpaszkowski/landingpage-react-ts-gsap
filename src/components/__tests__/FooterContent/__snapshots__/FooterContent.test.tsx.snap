// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <FooterContent /> component 1`] = `
.c2 > li {
  color: #E5E5E5;
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  list-style-type: none;
}

.c2 > li:first-child {
  color: #7C8176;
  margin-bottom: 1.9rem;
}

.c4 {
  position: relative;
  width: 35rem;
  height: 5rem;
  background-color: transparent;
  font-family: 'Reem Kufi',sans-serif;
  color: #7C8176;
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  border: 1px solid #7C8176;
  border-radius: 8px;
  padding: 1.2rem 14rem 1.2rem 1.6rem;
}

.c6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  font-family: 'Reem Kufi',sans-serif;
  font-size: 1.8rem;
  font-style: normal;
  font-weight: normal;
  line-height: 2.7rem;
  color: #131E07;
  border: none;
  border-radius: 4px;
  background-color: #FABD20;
  cursor: pointer;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  overflow: hidden;
  will-change: transform;
}

.c6::after {
  content: "";
  position: absolute;
  z-index: -1;
  bottom: -100%;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 4px;
  background-color: black;
  color: #FABD20;
  -webkit-transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
}

.c6:hover {
  box-shadow: 0px 1rem 2rem rgba(0,0,0,0.25);
  color: #FABD20;
}

.c6:hover::after {
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

.c3 {
  position: relative;
}

.c3 > p {
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  margin-bottom: 1.6rem;
  color: #7C8176;
}

.c7 {
  position: absolute;
  right: 0.8rem;
  bottom: 0.8rem;
  padding: 0.3rem 1.6rem 0.4rem 1.6rem;
  border-radius: 4px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin: 5.1rem 0 8rem 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 ul:nth-child(2) {
  margin-right: 14rem;
}

.c1 ul:first-child {
  margin-right: 5.8rem;
}

@media (max-width:375px) {
  .c2 {
    margin-bottom: 3.1rem;
  }
}

@media (max-width:375px) {
  .c4 {
    width: 100%;
  }
}

@media (max-width:1110px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:1110px) {
  .c7 {
    position: relative;
    width: 100%;
    right: 0;
    bottom: 0;
    font-size: 2rem;
    padding: 1.6rem 8rem 1.6rem 8rem;
    margin: 1.6rem 0 6.3rem 0;
  }
}

@media (max-width:1110px) {
  .c5 {
    width: 100%;
  }
}

@media (max-width:1110px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    margin: 5.6rem 0 8rem 0;
  }
}

@media (max-width:1110px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <ul
      className="c2"
    >
      <li>
        Products
      </li>
      <li>
        Updates
      </li>
      <li>
        Security
      </li>
      <li>
        Help
      </li>
    </ul>
    <ul
      className="c2"
    >
      <li>
        Company
      </li>
      <li>
        About
      </li>
      <li>
        Blog
      </li>
      <li>
        Join Us
      </li>
    </ul>
    <ul
      className="c2"
    >
      <li>
        Help
      </li>
      <li>
        Talk to Support
      </li>
      <li>
        Support Docs
      </li>
      <li>
        API Docs
      </li>
      <li>
        System Status
      </li>
    </ul>
  </div>
  <div
    className="c3"
  >
    <p>
      Try Medli for free
    </p>
    <div>
      <input
        className="c4 c5"
        placeholder="Enter your email"
      />
    </div>
    <button
      className="c6 c7"
      fontSize="1.8rem"
    >
      Get started
    </button>
  </div>
</div>
`;
